examples/tabs.rs

```diff
@@ -5,31 +5,72 @@ use crossterm::{
     execute,
     terminal::{disable_raw_mode, enable_raw_mode, EnterAlternateScreen, LeaveAlternateScreen},
 };
-use ratatui::{prelude::*, widgets::*};
+use ratatui::{prelude::*, style::palette::tailwind, widgets::*};
+use strum::{Display, EnumIter, FromRepr, IntoEnumIterator};
 
-struct App<'a> {
-    pub titles: Vec<&'a str>,
-    pub index: usize,
+const PALETTES: &[tailwind::Palette] = &[
+    tailwind::BLUE,
+    tailwind::EMERALD,
+    tailwind::INDIGO,
+    tailwind::RED,
+];
+
+const BORDER_TYPES: &[BorderType] = &[
+    BorderType::Rounded,
+    BorderType::Plain,
+    BorderType::Double,
+    BorderType::Thick,
+];
+
+#[derive(Default, Clone, Copy, Display, FromRepr, EnumIter)]
+enum SelectedTab {
+    #[default]
+    Tab0,
+    Tab1,
+    Tab2,
+    Tab3,
+}
+
+impl SelectedTab {
+    /// Get the previous tab, if there is no previous tab return the current tab.
+    fn previous(&self) -> Self {
+        let current_index: usize = *self as usize;
+        let previous_index = current_index.saturating_sub(1);
+        Self::from_repr(previous_index).unwrap_or(*self)
+    }
+
+    /// Get the next tab, if there is no next tab return the current tab.
+    fn next(&self) -> Self {
+        let current_index = *self as usize;
+        let next_index = current_index.saturating_add(1);
+        Self::from_repr(next_index).unwrap_or(*self)
+    }
+
+    /// Return enum name as a styled `Line` with two spaces both left and right.
+    fn tab_title(value: SelectedTab) -> Line<'static> {
+        let text = format!("  {value}  ");
+        let palette = &PALETTES[value as usize];
+        text.fg(tailwind::SLATE.c200).bg(palette.c900).into()
+    }
```

Comment:
This would make sense as `impl From<SelectedTab> for Line`.

Reply:
Thanks!

Reply:
Just a quick rationale for why I moved away from that implementation in the flex / constraints example. Naming it with "to_title" made the method describe the intended usage of the Line we're returning (Line being an implementation detail, but title being the core concept).